continuationsPlugin/src/main/scala/continuations/ContinuationsPlugin.scala

```diff
@@ -74,3 +96,141 @@ class ContinuationsPhase extends PluginPhase:
     tree match
       case Apply(_, _) => returnsContextFunctionWithSuspendType(tree)
       case _ => false
+
+  def hasOnlySuspendParam(tree: DefDef)(using Context): Boolean =
+    tree.termParamss match
+      case List(Nil, param :: Nil) => isSuspendType(param.tpe)
+      case _ => false
+
+  def getAllSubTrees(tree: Tree)(using Context): List[Tree] = {
+    val deepFolder: DeepFolder[ListBuffer[tpd.Tree]] =
+      new DeepFolder[ListBuffer[Tree]]((subTrees, tree) => subTrees += tree)
+
+    /*
+     * Needed for `case Inlined` because DeepFolder.foldOver for Inlined doesn't use `call` tree so the Inlined
+     * is not being unwrapped (???)
+     */
```

Comment:
Not a dotty expert (yet), but you may be able to avoid this by forcing the rhs tree of the DefDef, and get away with a `TreeCopier` [instead](https://github.com/lampepfl/dotty/blob/63344e7337727f886d3f7f1a0ad0234dd7670df8/compiler/src/dotty/tools/dotc/ast/Trees.scala#L1131-L1365) to selectively replace the statements in the tree via copying. I don't know if that's the standard way of doing it, but see the [LambdaLift](https://github.com/lampepfl/dotty/blob/a1157db1e44f23b1f1e605cf40ca2c0e88265c87/compiler/src/dotty/tools/dotc/transform/LambdaLift.scala#L192-L202) phase as an example.

Reply:
as said above, we don't have this method anymore